### PR TITLE
Prettier + lint checks in git hooks

### PR DESCRIPTION
**Pull request by pettinarip**
Things addressed here:

1. Added new scripts in the root of the app:
	1. `yarn format`: formats everything
	2. `yarn test:format`: checks if the format is ok or not
2. Autoformat files being added to the commit in the pre-commit hook (you can see this in the `.husky` folder).

**Bonus**:
I've added one more thing: running the tests and linting checks on the pre-push hook. 
- With this, we can catch any potential error before the CI or Netlify build system does.
- Might be a bit strange but I'm running only the vue tests for this (if you look into the pre-push script), not the `/contract` ones since those tests are too many and it takes a lot of time to run them. And since we are not doing that many changes in that project I think it does not deserve to do them on every push.
- Now the test script on the Vue app does not only run the unit tests but also runs the linting checks.

**TODO**:
- Run the autoformat on the entire codebase with `yarn format`. I will run this after we reach an agreement with this proposal. I don't want to add noise to the PR yet so that you can read clearly the important parts first.

***

I'm open to opinions of course, especially with the bonus point. Perhaps you want to leave that point to another time and within a different PR.

### CODE REVIEW

File: vue-app/.eslintrc.js
Comment (pettinarip) [inline vue-app/.eslintrc.js:30]: I've removed these rules since they are going to be handled by prettier.

**Review by samajammin — APPROVED** [review]

**Comment by samajammin** [comment]
Awesome, nice work!

> I've added one more thing: running the tests and linting checks on the pre-push hook.

I like this 😄 

> Might be a bit strange but I'm running only the vue tests for this

I'm ok with this - we don't change the contracts often.

> Run the autoformat on the entire codebase with yarn format. I will run this after we reach an agreement with this proposal. I don't want to add noise to the PR yet so that you can read clearly the important parts first.

Makes sense. Thanks!

**Review by corwintines — APPROVED** [review]

**Comment by samajammin** [comment]
I'll wait for further feedback on #364 before merging.

**Comment by pettinarip** [comment]
Yes! let's wait until we have more feedback.

**Pull request merged**
